src/gui/plugins/apply_force_torque/ApplyForceTorque.cc

```diff
@@ -0,0 +1,318 @@
+    }
+  }
+
+  // Get the position of the center of mass and link orientation
+  if (this->dataPtr->selectedEntity.has_value())
+  {
+    auto linkWorldPose = worldPose(*this->dataPtr->selectedEntity, _ecm);
+    auto inertial = _ecm.Component<components::Inertial>(
+      *this->dataPtr->selectedEntity);
+    if (inertial)
+    {
+      this->dataPtr->inertialPos =
+        linkWorldPose.Rot().RotateVector(inertial->Data().Pose().Pos());
+      this->dataPtr->linkRot = linkWorldPose.Rot();
+    }
+  }
+
+  emit this->ModelNameChanged();
+  emit this->LinkNameListChanged();
+  emit this->LinkIndexChanged();
+}
+
+/////////////////////////////////////////////////
+QString ApplyForceTorque::ModelName() const
+{
+  return this->dataPtr->modelName;
+}
+
+/////////////////////////////////////////////////
+QStringList ApplyForceTorque::LinkNameList() const
+{
+  return this->dataPtr->linkNameList;
+}
+
+/////////////////////////////////////////////////
+int ApplyForceTorque::LinkIndex() const
+{
+  return this->dataPtr->linkIndex;
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::SetLinkIndex(int _linkIndex)
+{
+  this->dataPtr->linkIndex = _linkIndex;
+  this->dataPtr->changedIndex = true;
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::UpdateForce(double _x, double _y, double _z)
+{
+  this->dataPtr->force = {_x, _y, _z};
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::UpdateTorque(double _x, double _y, double _z)
+{
+  this->dataPtr->torque = {_x, _y, _z};
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::ApplyForce()
+{
+  this->dataPtr->PublishWrench(true, false);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::ApplyTorque()
+{
+  this->dataPtr->PublishWrench(false, true);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::ApplyAll()
+{
+  this->dataPtr->PublishWrench(true, true);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorquePrivate::PublishWrench(bool _applyForce, bool _applyTorque)
+{
+  std::lock_guard<std::mutex> lock(this->mutex);
+
+  if (!this->selectedEntity.has_value())
+  {
+    gzdbg << "No link selected" << std::endl;
+    return;
+  }
+
+  // Force and torque in world coordinates
+  math::Vector3d forceToApply = this->linkRot.RotateVector(
+    _applyForce ? this->force : math::Vector3d::Zero);
+  math::Vector3d torqueToApply = this->linkRot.RotateVector(
+    _applyTorque ? this->torque : math::Vector3d::Zero) +
+    this->inertialPos.Cross(forceToApply);
+
+  msgs::EntityWrench msg;
+  msg.mutable_entity()->set_id(*this->selectedEntity);
+  msgs::Set(msg.mutable_wrench()->mutable_force(), forceToApply);
+  msgs::Set(msg.mutable_wrench()->mutable_torque(), torqueToApply);
+
+  this->pub.Publish(msg);
+}
+
+// Register this plugin
+GZ_ADD_PLUGIN(ApplyForceTorque, gz::gui::Plugin);
```

Comment:
The `;` is causing a warning.
```suggestion
GZ_ADD_PLUGIN(ApplyForceTorque, gz::gui::Plugin)
```

Reply:
Done in 3911d73